web-api/src/storage/elastic.rs

```diff
@@ -623,8 +603,10 @@ impl NormalizationFn {
     }
 }
 
+type DynMergeFn = dyn Fn(ScoreMap<DocumentId>, ScoreMap<DocumentId>) -> ScoreMap<DocumentId>;
+
 impl MergeFn {
-    fn to_fn(self) -> Box<dyn Fn(ScoreMap, ScoreMap) -> ScoreMap> {
+    fn to_fn(self) -> Box<DynMergeFn> {
```

Comment:
❓ since the type alias is not reused elsewhere we could just inline it here. or is the linter complaining about type complexity?

```suggestion
impl MergeFn {
    fn to_fn(self) -> Box<dyn Fn(ScoreMap<DocumentId>, ScoreMap<DocumentId>) -> ScoreMap<DocumentId>> {
```

Reply:
> or is the linter complaining about type complexity?

yes